framework/audio/iaudioinsert.h

```diff
@@ -0,0 +1,46 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_IAUDIOINSERT_H
+#define MU_AUDIO_IAUDIOINSERT_H
+
+namespace mu {
+namespace audio {
+class IAudioInsert
+{
+public:
+    virtual ~IAudioInsert() = default;
+
+    virtual void setSampleRate(unsigned int sampleRate) { m_sampleRate = sampleRate; }
+
+    virtual bool active() const { return m_active; }
+    virtual void setActive(bool active) { m_active = active; }
+
+    //! return channel count for this insert: 1 for mono, 2 for stereo
+    virtual unsigned int streamCount() const = 0;
```

Comment:
Seems that this method should return enum and its name should be different

Reply:
No. It is a count of audio streams. 1 or 2 is most popular values. But it is also possible that audio stream will have 5 or 6 or other count of streams.

Reply:
Why does this method have "streamCount" name if this comment says another (channel count)?